includes/admin/class-sensei-course-pre-publish-panel.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * File containing the class Sensei_Course_Pre_Publish_Panel.
+ *
+ * @package sensei-lms
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class that handles the pre-publish panel for courses.
+ *
+ * @since 4.5.0
+ */
+class Sensei_Course_Pre_Publish_Panel {
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_Course_Pre_Publish_Panel constructor. Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {
+	}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_pre_publish_panel_assets' ) );
+		add_action( 'publish_course', array( $this, 'maybe_publish_lessons' ) );
+	}
+
+	/**
+	 * Enqueue pre-publish panel assets.
+	 */
+	public function enqueue_pre_publish_panel_assets() {
+		if ( 'course' !== get_post_type() ) {
+			return;
+		}
+
+		Sensei()->assets->enqueue( 'sensei-course-pre-publish-panel-script', 'admin/course-pre-publish-panel/index.js' );
+	}
+
+	/**
+	 * Maybe publish associated lessons when the course is published.
+	 *
```

Comment:
Should we add an `internal` tag here?

Reply:
https://github.com/Automattic/sensei/pull/7536/commits/2c5b426ed3b0ae258baf3e074814522e5678d66e